docs/shared/recipes/running-tasks/customizing-inputs.md

```diff
@@ -0,0 +1,354 @@
+
+{% tabs %}
+{% tab label="Global" %}
+
+```json {% fileName="nx.json" highlightLines=[4] %}
+{
+  "targetDefaults": {
+    "build": {
+      "inputs": ["..."]
+    }
+  }
+}
+```
+
+{% /tab %}
+{% tab label="Project Level (project.json)" %}
+
+```json {% fileName="packages/some-project/project.json" highlightLines=[6] %}
+{
+  "name": "some-project",
+  ...
+  "targets": {
+    "build": {
+      "inputs": ["..."],
+      ...
+    }
+    ...
+  }
+}
+```
+
+{% /tab %}
+
+{% tab label="Project Level (package.json)" %}
+
+```json {% fileName="packages/some-project/package.json" highlightLines=[9] %}
+{
+  "name": "some-project",
+  "dependencies": {},
+  "devDependencies": {},
+  ...
+  "nx": {
+    "targets": {
+      "build": {
+        "inputs": ["..."],
+        ...
+      }
+      ...
+    }
+  }
+}
+```
+
+{% /tab %}
+
+## Include all project files and dependencies in cache hash
+
+The following definition includes all files of the project itself as well as of all its dependencies in the cache hash, hence telling Nx to invalidate the cache whenever
+
+- any file of the project itself changes
+- any file of any dependency changes
+
+```jsonc {% fileName="nx.json" %}
+{
+  "namedInputs": {
+    "default": ["{projectRoot}/**/*"]
+  },
+  "targetDefaults": {
+    "build": {
+      "inputs": ["default", "^default"]
+    }
+  }
+}
+```
+
+This definition is the default behavior of Nx, even if you don't specify any `inputs` at all. This is a rather cautious approach. It will always give you correct output, but it might re-run a task in some cases where the cache could have been used instead.
+
+Note, Nx uses the [minimatch](https://github.com/isaacs/minimatch) library to process glob patterns.
+
+{% callout title="Replacements" %}
+
+Note how you can use `{projectRoot}` and `{workspaceRoot}` as placeholders to simplify writing glob definitions.
+
+{% /callout %}
+
+If you're wondering what `namedInputs` are, read the next section.
+
+## Reusing inputs definitions with namedInputs
+
+If you find yourself reusing the same `inputs` definitions, you can instead create a `namedInput`. It is like a variable definition which can then be reused in the `inputs` array.
+
+```jsonc {% fileName="nx.json" highlightLines=[3, 7] %}
+{
+  "namedInputs": {
+    "default": ["{projectRoot}/**/*"]
+  },
+  "targetDefaults": {
+    "build": {
+      "inputs": ["default", "^default"]
+    }
+  }
+}
+```
+
+The `^` character at the beginning of the `^default` string means this entry applies to the project dependencies of the project, not the project itself. In our example of `myreactapp` depending on `shared-ui`, if we run
+
+```shell
+nx build myreactapp
+```
+
+...then by having defined
+
+- `: ["default",...]` - it will invalidate the cache of `myreactapp` whenever some src file of `myreactapp` itself changes
```

Comment:
```suggestion
- `"inputs": ["default",...]` - it will invalidate the cache of `myreactapp` whenever some src file of `myreactapp` itself changes
```